src/mem/mod.rs

```diff
@@ -9,3 +9,25 @@ mod arena;
 pub use arena::*;
 
 pub mod cow;
+
+/// Aligns the given address to the alignment for the given type.
+///
+/// `align` must be a power of two; otherwise, the returned value
+/// will be well-defined but unspecified.
+///
+/// This function will always return a value greather than or equal to `addr`.
+#[inline]
+pub(in crate) fn align_to(addr: usize, align: usize) -> usize {
+    let mask = align.saturating_sub(1);
+    let (addr, overflow) = addr.overflowing_add(mask);
+    if overflow {
+        return usize::MAX;
```

Comment:
You may want to mention that in the case of overflow, this will return an unaligned value. I think returning `usize::MAX` is a bit of a footgun, although my only suggestion is "document it better".